rust/lance-index/src/vector/hnsw.rs

```diff
@@ -213,13 +217,37 @@ struct SchemaIndexMetadata {
     metric_type: String,
 }
 
-#[derive(Debug, Serialize, Deserialize)]
+#[derive(Debug, Clone, Serialize, Deserialize)]
 pub struct HnswMetadata {
     entry_point: u32,
     level_offsets: Vec<usize>,
 }
 
 impl HNSW {
+    pub fn empty() -> Self {
+        Self {
+            levels: vec![],
+            metric_type: MetricType::L2,
+            entry_point: 0,
+            use_select_heuristic: true,
+        }
+    }
+
+    /// The number of nodes in the graph.
```

Comment:
Is this the # of nodes in the graph?  Or just in the top level of the graph?

Reply:
it's the # of nodes in the level 0